security/monitoring.adoc

```diff
@@ -0,0 +1,191 @@
+
+4
+----
+
+[NOTE]
+====
+When a namespace is deleted, its events are deleted as well. Events can also expire and are deleted to prevent filling up *etcd* storage. Events are
+not stored as a permanent record and frequent polling is necessary to capture statistics over time.
+====
+
+[[security-monitoring-logs]]
+== Cluster Logs
+
+This section describes the types of operational logs produced on the cluster.
+
+[[security-monitoring-service-logs]]
+=== Service Logs
+
+{product-title} produces logs with each *systemd* service that is running on a host:
+
+ifdef::openshift-origin[]
+- origin-master-api
+- origin-master-controllers
+- etcd
+- origin-node
+endif::[]
+ifdef::openshift-enterprise[]
+- atomic-openshift-master-api
+- atomic-openshift-master-controllers
+- etcd
+- atomic-openshift-node
+endif::[]
+
+These logs are intended more for debugging purposes than for security auditing. They can be
+retrieved per host with `journalctl` or, in clusters with the aggregated logging stack deployed, in the logging _.operations_ indexes (possibly
+in the xref:../install_config/aggregate_logging.adoc#aggregated-ops[Ops cluster]) as a cluster administrator.
+
+[[security-monitoring-audit-log]]
+=== Master API Audit Log
+
+To log master API requests by users, administrators, or system components enable audit
+logging xref:../install_config/master_node_configuration.adoc#master-node-config-audit-config[for the master API]. This will
+create a file on each master host or, if there is no file configured, be included in the service's journal. Entries in the journal
+can be found by searching for *"AUDIT"*.
+
+xref:../install_config/master_node_configuration.adoc#master-node-config-audit-config[Audit
+log entries] consist of one line recording each REST request when it is
+received and one line with the HTTP response code when it completes. For
+example, here is a record of the system administrator requesting a list
+of nodes:
+
+----
+2017-10-17T13:12:17.635085787Z AUDIT: id="410eda6b-88d4-4491-87ff-394804ca69a1" ip="192.168.122.156" method="GET" user="system:admin" groups="\"system:cluster-admins\",\"system:authenticated\"" as="<self>" asgroups="<lookup>" namespace="<none>" uri="/api/v1/nodes"
+2017-10-17T13:12:17.636081056Z AUDIT: id="410eda6b-88d4-4491-87ff-394804ca69a1" response="200"
+----
+
+It might be useful to poll the log periodically for the number of recent requests per response code, as shown in the following example:
+
+----
+$ tail -5000 /var/log/openshift-audit.log \
+  | grep -Po 'response="..."' \
+  | sort | uniq -c | sort -rn
+
+   3288 response="200"
+      8 response="404"
+      6 response="201"
+----
+
+The following list describes some of the response codes in more detail:
+
+* `200` or `201` response codes indicate a successful request.
+* `400` response codes may be of interest as they indicate a malformed request, which should not occur with most clients.
+* `401` and `403` response codes are simple redirects.
```

Comment:
Wow, I don't know where my brain went on this. The return codes I meant here should be 301 and 302. But there's probably no real reason to mention these at all (I must have been thinking the 4XX range would be confusing with some codes "normal" and some "suspicious"). So I'd say just delete this line. My mistake!